### PR TITLE
Validate your YAML (CUE)

### PR DESCRIPTION
**Pull request by adamgordonbell**
Here is my CUE tutorial:

https://62a20f91a0bc8917cd49068f--ephemeral-narwhal-3ac3b2.netlify.app/blog/yaml-validate-and-lint-cue-lang/

### CODE REVIEW

File: blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+
+That's a small win for `cue`, but it shows how static typing can be valuable.
+
+<div class="notice--big--primary">
+
+## What Did I Learn?
+
+Cuelang can be used to specify a schema for a plain YAML file. You can put your types in a separate file and run `cue vet types.cue plain.yml` and start benefiting from static types right away.
+
+To specify a Type, the convention is to with a `#` like this:
```
Comment (vladaionescu) [inline blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md:112]:
```suggestion
To specify a Type, the convention is to prefix with a `#` like this:
```

File: blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+
+The Cuelang standard library has [many other](https://pkg.go.dev/cuelang.org/go@v0.4.3/pkg) functions for constraining types. Any function that takes in the type in question and returns a bool can be used for this purpose. In this way, constraints in CueLang are like [Refinement Types](https://en.wikipedia.org/wiki/Refinement_type).
+
+The number of potential invariants you can enforce in this way is immense. For example, I could parse the links to ensure they are valid links, and I could ensure that `avatar` string always ended in `.png` and so on. However, for this simple author list, what I've got is enough validation.
+
+<div class="notice--big--primary">
+
+## What I Learned
+
+Cuelang lets you constrain your values by adding constraints to the types. There are built-in constraints that let enforce ranges and ensure strings match or don't match regular expressions like this:
```
Comment (vladaionescu) [inline blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md:296]:
```suggestion
Cuelang lets you constrain your values by adding constraints to the types. There are built-in constraints that let you enforce ranges and ensure strings match or don't match regular expressions like this:
```

File: blog/_data/authors.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ Josh:
...
-    bio     : "Spreading the word about Earthly. Host of CoRecursive podcast. Physical Embodiment of Cunningham's Law"
+    bio     : "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eu dolor nec nisi scelerisque semper. Phasellus tincidunt ut leo et imperdiet. Etiam dignissim orci non egestas feugiat. Ut lacinia sem quis nisi tincidunt, a pellentesque lectus luctus. Proin hendrerit laoreet lectus in euismod. Donec sed mauris in libero luctus semper. Proin ornare tortor sit amet velit placerat mattis. Nam et interdum mauris, vel consequat enim. Ut ut libero urna. Cras viverra convallis velit sit amet porta. Vestibulum in quam at nibh pretium ultrices ut sed risus. Curabitur suscipit, lacus et sodales rutrum, magna augue vehicula arcu, vitae feugiat leo nisl sit amet risus. Nullam a neque quam. Fusce luctus, lectus nec scelerisque pretium, tellus lectus maximus lectus, in pretium quam risus sed augue. Donec fringilla imperdiet enim, eget interdum neque varius in. Quisque elementum libero vel massa volutpat varius. Nam faucibus neque id nibh vulputate, et placerat libero tincidunt. Interdum et malesuada fames ac ante ipsum primis in faucibus."
```
Comment (vladaionescu) [inline blog/_data/authors.yml:27]: You might want to undo this - although it is a little funny to actually see this bio at the end.

Screen Shot 2022-06-10 at 10 09 56 AM

**Review by vladaionescu — APPROVED** [review]
:+1:

Looks cool! Minor comments...

File: blog/_data/authors.yml (hunk 3 above)
Comment (adamgordonbell) [reply blog/_data/authors.yml:27]: Ha, i forgot about that.

File: blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+
+~~~{.bash caption=">_"}
+$ cue vet authors.cue
+~~~
+
+And I can be sure all my config data is structured correctly.
+
+Once you start import types as part of config writing, you can see why something like CueLang or its competitor Dhall can make a lot of sense. Especially for config-heavy domains like Kubernetes.
+
+(A great post the expands on this idea is [How CUE wins](https://blog.cedriccharly.com/post/20210523-how-cue-wins/) which makes a great case for why something like CUE is needed desperately needed.)
```
Comment (jalletto) [inline blog/_posts/2022-06-07-yaml-validate-and-lint-cue-lang.md:447]:
```suggestion
(A great post that expands on this idea is [How CUE wins](https://blog.cedriccharly.com/post/20210523-how-cue-wins/) which makes a great case for why something like CUE is  desperately needed.)
```

**Review by jalletto — APPROVED** [review]
Not going to lie, before this I was kind of like, what the hell is cue though actually? This was a great and informative write up.

**Comment by vladaionescu** [comment]
> I think I was the ideal audience for this article because I only very vaguely understood what cue was before reading this. Great break down. I totally get it now.

Same here!

**Pull request merged**